scripts/scr_company_view/scr_company_view.gml

```diff
@@ -144,51 +180,48 @@ function scr_company_view(company) {
 	                }*/
 
 	                if (ma_role[v]=obj_ini.role[100,5]){
-	                    if (company=10) and (ma_exp[v]>=40) then ma_promote[v]=1;
-	                    if (company=9) and (ma_exp[v]>=50) then ma_promote[v]=1;
-	                    if (company=8) and (ma_exp[v]>=60) then ma_promote[v]=1;
-	                    if (company=7) and (ma_exp[v]>=70) then ma_promote[v]=1;
-	                    if (company=6) and (ma_exp[v]>=80) then ma_promote[v]=1;
-	                    if (company=5) and (ma_exp[v]>=100) then ma_promote[v]=1;
-	                    if (company=4) and (ma_exp[v]>=110) then ma_promote[v]=1;
-	                    if (company=3) and (ma_exp[v]>=120) then ma_promote[v]=1;
-	                    if (company=2) and (ma_exp[v]>=150) then ma_promote[v]=1;
-
+	                    if (company==10) and (ma_exp[v]>=40) then ma_promote[v]=1;
+	                    else if (company==9) and (ma_exp[v]>=50) then ma_promote[v]=1;
+	                    else if (company==8) and (ma_exp[v]>=60) then ma_promote[v]=1;
+	                    else if (company==7) and (ma_exp[v]>=70) then ma_promote[v]=1;
+	                    else if (company==6) and (ma_exp[v]>=80) then ma_promote[v]=1;
+	                    else if (company==5) and (ma_exp[v]>=100) then ma_promote[v]=1;
+	                    else if (company==4) and (ma_exp[v]>=110) then ma_promote[v]=1;
+	                    else if (company==3) and (ma_exp[v]>=120) then ma_promote[v]=1;
+	                    else if (company==2) and (ma_exp[v]>=150) then ma_promote[v]=1;
 	                }
-
-	                if (obj_controller.command_set[2]=1) and (ma_promote[v]=0) then ma_promote[v]=1;
+	                if (obj_controller.command_set[2]==1) and (ma_promote[v]==0) then ma_promote[v]=1;
 	            }
 	        }
 
-	        if (obj_ini.name[company,v+1]="") and
-	        (obj_ini.name[company,v]!="") and
-	        (last_man=0) and
-	        (obj_ini.ship_location[obj_ini.lid[company,v]]!="Lost")
+	        if (obj_ini.name[company,v+1]=="") and (obj_ini.name[company,v]!="") and
+	        (last_man==0) and (obj_ini.ship_location[obj_ini.lid[company,v]]!="Lost")
 	        then last_man=v;
 	    }
 	}
 
-
-	var v=last_man;
+	v=last_man;
 	last_vehicle=0;
-
 	for (var i=1;i<=100;i++){
 	// if (!instance_exists(obj_popup)) then repeat(100){// 100
 		bad=0;
 
 	    // if (obj_ini.veh_race[company,i]=company) and (obj_ini.veh_role[company,i]!=""){
 	    if (obj_ini.veh_race[company,i]!=0){
 	        if (obj_ini.veh_lid[company,i]>0){
-	            if (obj_ini.ship_location[obj_ini.veh_lid[company,i]]="Lost") then bad=1;
+	            if (obj_ini.ship_location[obj_ini.veh_lid[company,i]]=="Lost") then bad=1;
```

Comment:
may as well just place a continue in rather than having to evaluate bad later. Optional though as the gins are minimal